lib/src/value/calculation.dart

```diff
@@ -120,6 +125,189 @@ class SassCalculation extends Value {
     return SassCalculation._("max", args);
   }
 
+  /// Creates a `hypot()` calculation with the given [arguments].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation]. It must be passed at least one argument.
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value hypot(Iterable<Object> arguments) {
+    var args = _simplifyArguments(arguments);
+    if (args.isEmpty) {
+      throw ArgumentError("hypot() must have at least one argument.");
+    }
+    _verifyCompatibleNumbers(args);
+
+    var subtotal = 0.0;
+    var first = args.first;
+    if (first is! SassNumber || first.hasUnit('%')) {
+      return SassCalculation._("hypot", args);
+    }
+    for (var i = 0; i < args.length; i++) {
+      var number = args.elementAt(i);
+      if (number is! SassNumber || !number.hasCompatibleUnits(first)) {
+        return SassCalculation._("hypot", args);
+      }
+      var value =
+          number.convertValueToMatch(first, "numbers[${i + 1}]", "numbers[1]");
+      subtotal += value * value;
+    }
+    return SassNumber.withUnits(math.sqrt(subtotal),
+        numeratorUnits: first.numeratorUnits,
+        denominatorUnits: first.denominatorUnits);
+  }
+
+  /// Creates a `sqrt()` calculation with the given [argument].
+  ///
+  /// The [argument] must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value sqrt(Object argument) =>
+      _singleArgument("sqrt", argument, number_lib.sqrt);
+
+  /// Creates a `sin()` calculation with the given [argument].
+  ///
+  /// The [argument] must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  static Value sin(Object argument) =>
+      _singleArgument("sin", argument, number_lib.sin,
+          requireUnits: true, requireKnownUnits: true);
```

Comment:
These trigonometric functions shouldn't require units. `sin(0.7)` is valid, and interprets its argument as radians.

Reply:
Hm that's I think the naming is confusing, `requireUnits` doesn't apply to any calculation function then because they all accept unit-less parameters. Renaming it to `forbidUnits` makes more sense now.